monkey/monkey_island/cc/repositories/plugin_utils.py

```diff
@@ -0,0 +1,73 @@
+import json
+from tarfile import TarFile
+from typing import Any, BinaryIO, Dict
+
+import yaml
+
+from common.agent_plugins import AgentPlugin, AgentPluginManifest
+
+from . import RetrievalError
+
+MANIFEST_FILENAME = "plugin.yaml"
+CONFIG_SCHEMA_FILENAME = "config-schema.json"
+SOURCE_ARCHIVE_FILENAME = "data.tar"
+
+
+def load_plugin(file: BinaryIO) -> AgentPlugin:
+    """
+    Load a plugin from a tar file.
+
+    :raises RetrievalError: if the plugin failed to load
+    """
+    tar_file = TarFile(fileobj=file)
+    manifest = get_plugin_manifest(tar_file)
+    schema = get_plugin_schema(tar_file)
+    source = get_plugin_source(tar_file)
+
+    return AgentPlugin(plugin_manifest=manifest, config_schema=schema, source_archive=source)
+
+
+def get_plugin_manifest(tar: TarFile) -> AgentPluginManifest:
+    """
+    Retrieve the plugin manifest from a tar file.
+
+    :raises KeyError: if the manifest was not found in the tar file
+    :raises RetrievalError: if the manifest could not be loaded
+    """
+    manifest_info = tar.getmember(MANIFEST_FILENAME)
+    manifest_buf = tar.extractfile(manifest_info)
+    if not manifest_buf:
+        raise RetrievalError("Error retrieving plugin manifest file")
```

Comment:
I'm not sure this component should be raising `RetrievalError`. That should perhaps be left up to the repository. Perhaps `ValueError` if the `TarFile` is missing data.